### PR TITLE
make elasticsearch provider optional if search.elasticsearch not defined

### PR DESCRIPTION
**Pull request by aochsner**
## Hey, I just made a Pull Request!

For the new backend, make adding the elasticsearch search provider optional if `search.elasticsearch` is not defined in the config

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @aochsner, at a high level this makes sense to me, will leave the final say to the @discoverability-maintainers.

I did leave a pair of comments that should be pretty easy to address 👍

File: plugins/search-backend-module-elasticsearch/src/alpha.ts
Comment (aochsner) [reply plugins/search-backend-module-elasticsearch/src/alpha.ts:71]: Yep makes sense... how's that look for a message?

**Comment by aochsner** [comment]
> Thanks for this @aochsner, at a high level this makes sense to me, will leave the final say to the @discoverability-maintainers.
> 
> I did leave a pair of comments that should be pretty easy to address 👍

Thanks for the feedback!  Incorporated.

File: plugins/search-backend-module-elasticsearch/src/alpha.ts
Comment (awanlin) [inline plugins/search-backend-module-elasticsearch/src/alpha.ts:74]:
```suggestion
          logger.warn('No configuration found under "search.elasticsearch" key.  Skipping search engine inititalization.');
```

Feel like warning is better, will jump out at them they missed something.

File: plugins/search-backend-module-elasticsearch/src/alpha.ts
Comment (awanlin) [reply plugins/search-backend-module-elasticsearch/src/alpha.ts:71]: Yup! Thanks

**Review by awanlin — CHANGES_REQUESTED** [review]
Looks good, just a small adjustment for the log.

File: plugins/search-backend-module-elasticsearch/src/alpha.ts
Comment (aochsner) [reply plugins/search-backend-module-elasticsearch/src/alpha.ts]: Done!  Fixed a prettier check as well

**Review by awanlin — APPROVED** [review]
Thanks again for this @aochsner, looks good to me but the @discoverability-maintainers will need to review as well and they may have a different view.

**Review by freben — APPROVED** [review]
i think this is fine to ship - @emmaindal ?

**Review by emmaindal — APPROVED** [review]
Yup I'm fine with this, thank you!

**Pull request merged**
